### DISCUSSION TITLE
Incorrect reading of cumulative electricity costs

### DISCUSSION THREAD

**Discussion by judgepd** [BottlecapDave/HomeAssistant-OctopusEnergy#934] [Q&A]
My zappi was in use during the day yesterday and although the cumulative electricity consumption was going up as expected, the cost seems to be dropping each half an hour and then climbing again meaning that the total value for the day is much lower than it really is.

Does anyone have any ideas what might be happening?

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/94126799/40ed7b97-18e6-4013-8f17-8bb0de3d6fc2)
![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/94126799/6271e320-1601-43c4-b096-a7ec6be1d1b6)

**Comment by BottlecapDave** [comment]
Assuming you're on intelligent tariff and was allowing OE to find charge spots Vs bump charges, my _guess_ would be a delay between OE removing a pending dispatch and providing a completed dispatch in its place. The rates need to be adjusted based on the presence of these records as the basic API doesn't take them into account.

**Reply by judgepd** [reply]
Thanks for the reply. I am on intelligent, but for that session I had switched off smart charging and the zappi was just on fast.

**Reply by BottlecapDave** [reply]
It might be then that OE is still providing pending dispatches despite smart charging being turn off. If you have a chance when you next do this could you keep an eye on the pending dispatches in the dispatching sensor and see if ones still come through?

**Comment by judgepd** [comment]
I'm not sure if this helps or not.  I charged yesterday using Intelligent Octopus Go with smart charging enabled and got continuous cheap slots starting at 20:30.  The cost seemed to be tracking correctly (and ultimately was) but there was a sudden jump up and drop down.  I did notice that although the dispatching sensor showed as 'On' the whole time, in the Octopus app it showed as slots from 20:30 - 21:00 and then 21:00 to 05:30

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/94126799/3798346f-6f2c-4edc-9989-79a3b7b23e4b)
